java/vector/src/main/java/org/apache/arrow/vector/TypeLayout.java

```diff
@@ -225,6 +225,117 @@ public TypeLayout visit(Duration type) {
     return layout;
   }
 
+  /**
+   * Gets the number of {@link BufferLayout}s for the given <code>arrowType</code>.
+   */
+  public static int getTypeBufferCount(final ArrowType arrowType) {
+    return  arrowType.accept(new ArrowTypeVisitor<Integer>() {
+
+      static final int FIXED_WIDTH_BUFFER_COUNT = 2;
+
+      static final int VARIABLE_WIDTH_BUFFER_COUNT = 3;
+
+      @Override
+      public Integer visit(Int type) {
+        return FIXED_WIDTH_BUFFER_COUNT;
+      }
+
+      @Override
+      public Integer visit(Union type) {
+        switch (type.getMode()) {
+          case Dense:
+            // TODO: validate this
+            return 3;
+          case Sparse:
+            return 1;
+          default:
+            throw new UnsupportedOperationException("Unsupported Union Mode: " + type.getMode());
+        }
+      }
+
+      @Override
+      public Integer visit(Struct type) {
+        return 1;
+      }
+
+      @Override
+      public Integer visit(Timestamp type) {
+        return FIXED_WIDTH_BUFFER_COUNT;
+      }
+
+      @Override
+      public Integer visit(org.apache.arrow.vector.types.pojo.ArrowType.List type) {
+        return 2;
```

Comment:
Would be good to make a note of buffers. For example, in case of LIST there is 1 validity and 1 offset.

Reply:
Revised. Thank you.